### PR TITLE
build: Fix clang errors

### PR DESCRIPTION
**Pull request by ncesario-lunarg**

### CODE REVIEW

File: layers/core_validation.cpp [hunk 1]
```diff
@@ -7662,8 +7662,6 @@ bool CoreChecks::ValidateEventStageMask(const ValidationStateTracker *state_data
     return skip;
 }
 
-static const VkQueueFlagBits kQueueTypeArray[] = {VK_QUEUE_GRAPHICS_BIT, VK_QUEUE_COMPUTE_BIT, VK_QUEUE_TRANSFER_BIT};
```
Comment (ncesario-lunarg) [inline layers/core_validation.cpp:7665]: I verified offline with @jeremyg-lunarg this should be removed.

File: layers/synchronization_validation.cpp [hunk 2]
```diff
@@ -5099,7 +5099,7 @@ bool SyncOpPipelineBarrier::Validate(const CommandBufferAccessContext &cb_contex
     assert(context);
     if (!context) return skip;
     // Validate Image Layout transitions
-    for (const auto image_barrier : image_memory_barriers_) {
+    for (const auto &image_barrier : image_memory_barriers_) {
```
Comment (ncesario-lunarg) [inline layers/synchronization_validation.cpp:5102]: Warning about copying data. I'm guessing a copy is not what's desired here?

File: layers/synchronization_validation.h [hunk 3]
```diff
@@ -235,6 +235,7 @@ class ResourceAccessState : public SyncStageAccess {
         VkPipelineStageFlags exec_scope;
         SyncStageAccessFlags access_scope;
         OrderingBarrier() = default;
+        OrderingBarrier(VkPipelineStageFlags es, SyncStageAccessFlags as) : exec_scope(es), access_scope(as) {}
```
Comment (ncesario-lunarg) [inline layers/synchronization_validation.h:238]: This one is technically not necessary for C++11, but clang 11 and MSVC (2019) complain if this doesn't exist when the standard is >C++11.

Comment (jeremyg-lunarg) [reply layers/synchronization_validation.h:238]: that is weird, but looks ok.

File: layers/synchronization_validation.cpp (hunk 2 above)
Comment (jeremyg-lunarg) [reply layers/synchronization_validation.cpp:5102]: no need for copy here.

File: layers/core_validation.cpp (hunk 1 above)
Comment (jeremyg-lunarg) [reply layers/core_validation.cpp:7665]: doeth.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

File: layers/synchronization_validation.h (hunk 3 above)
Comment (ncesario-lunarg) [reply layers/synchronization_validation.h:238]: I agree. [This](https://stackoverflow.com/questions/14178264/c11-correct-stdarray-initialization) SO post gives some insight I think.

**Pull request merged**